test/prototype/test_parq.py

```diff
@@ -265,8 +276,70 @@ def test_intx_weight_only_e2e(self, b: int = 2, group_size: int = 32):
         self.compare_parq_convert(model, m_ref, optimizer, config)
 
 
+class TestInt8DynamicActivationTorchaoQuantizer(common_utils.TestCase):
+    def setUp(self):
+        torch.manual_seed(123)
+
+    @unittest.skipIf(not TORCH_VERSION_AT_LEAST_2_6, "Test only enabled for 2.6+")
+    @common_utils.parametrize("b", [2, 3, 4, 8])
+    @common_utils.parametrize("model_dtype", [torch.float16, torch.float32])
+    @common_utils.parametrize("group_size", [32, 128])
+    def test_int8_dynamic_activation_intx_e2e(
+        self,
+        b: int = 2,
+        model_dtype: torch.dtype = torch.float32,
+        group_size: int = 32,
+    ):
+        model = M(embedding=False).to(_DEVICE, dtype=model_dtype)
+        x = model.example_inputs(device=_DEVICE).to(model_dtype)
+
+        # reference model using native quantization
+        m_ref = copy.deepcopy(model).eval().to(_DEVICE)
+        quantizer = UnifTorchaoQuantizer()
+        config = Int8DynamicActivationIntxWeightConfig(
+            weight_dtype=_BIT_WIDTH_TO_DTYPE[b],
+            weight_granularity=PerGroup(group_size),
+            weight_mapping_type=quantizer.mapping_type,
+            act_mapping_type=MappingType.ASYMMETRIC,
+        )
+        quantize_(m_ref, config)
+        ref_out = m_ref(x)
+
+        # quantize weights with PARQ
+        base_optimizer = torch.optim.SGD(build_param_groups(model, b, group_size))
+        optimizer = QuantOptimizer(
+            base_optimizer, quantizer, ProxHardQuant(), quant_per_channel=True
+        )
+        optimizer.zero_grad()
+        optimizer.step()
+
+        # apply torchao quantized activations on top
+        activation_config = FakeQuantizeConfig(
+            torch.int8,
+            granularity="per_token",
+            mapping_type=config.act_mapping_type,
+        )
+        filter_fn = optimizer.get_filter_fn(model)
+        quantize_(
+            model,
+            IntXQuantizationAwareTrainingConfig(activation_config=activation_config),
+            filter_fn=filter_fn,
+        )
+        out = model(x)
+        self.assertTrue(out.equal(ref_out))
+
+        # equivalent to torchao's convert step
+        model.eval()
+        optimizer.restore_latent_params()
+        quantize_(model, FromIntXQuantizationAwareTrainingConfig(), filter_fn=filter_fn)
+        quantize_(model, config, filter_fn=filter_fn)
+        converted_out = model(x)
+        self.assertTrue(converted_out.equal(ref_out))
```

Comment:
You can use `torch.testing.assert_close(converted_out, ref_out)` so if it fails it'll have a more detailed message saying how much of the tensor doesn't match